addons/mail/static/src/messaging/entity/entity/entity.js

```diff
@@ -0,0 +1,800 @@
+odoo.define('mail.messaging.entity.Entity', function (require) {
+'use strict';
+
+const { registerNewEntity } = require('mail.messaging.entity.core');
+
+const STORED_RELATION_PREFIX = `_`;
+
+function EntityFactory() {
+
+    class Entity {
+
+        /**
+         * @static
+         * @param {Object} [data]
+         */
+        static init(data) {
+            const { state } = this.env.store;
+            this.localId = this.name;
+            state.entities[this.localId] = this;
+            /**
+             * Classes are functions, and changes in functions are not observed
+             * by OWL. The work-around is to store data and relations of classes
+             * in an object, and make this object observed from store state.
+             */
+            state.__classEntityObservables[this.localId] = {};
+            this.observable = state.__classEntityObservables[this.localId];
+
+            // Class stored relation access (redirect to observable)
+            for (const relationName in this.relations) {
+                Object.defineProperty(this, this.__getStoredRelationName(relationName), {
+                    get: () => this.observable[this.__getStoredRelationName(relationName)],
+                });
+            }
+
+            // Class attribute access (redirect to observable)
+            const classAttributeNames = this._getListOfClassAttributeNames();
+            for (const classAttributeName of classAttributeNames) {
+                Object.defineProperty(this, classAttributeName, {
+                    get: () => this.observable[classAttributeName],
+                });
+            }
+
+            this.__init(this);
+            this.__update(this, data);
+        }
+
+        //--------------------------------------------------------------------------
+        // Public
+        //--------------------------------------------------------------------------
+
+        /**
+         * @static
+         * @returns {mail.messaging.entity.Entity[]} all instances of this entity
+         */
+        static get all() {
```

Comment:
add purpose of method (kinda obvious but this is the core class so it has to be well documented)